### PR TITLE
Enable enr field in GetPeers and GetPeerById (#8641)

### PR DESCRIPTION
**Pull request by zilm13**
Cherry-picked from master

### CODE REVIEW

File: data/provider/src/main/java/tech/pegasys/teku/api/peer/Eth2PeerWithEnr.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.api.peer;
+
+import java.util.Optional;
+import tech.pegasys.teku.networking.eth2.peers.Eth2Peer;
+
+public record Eth2PeerWithEnr(Eth2Peer peer, Optional<String> enr) {}
```
Comment (Nashatyrev) [inline data/provider/src/main/java/tech/pegasys/teku/api/peer/Eth2PeerWithEnr.java:19]: As the `enr` is optional, does it make sense to just add this field to `EthPeer`?

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/peers/DiscoveryNodeIdExtractor.java [hunk 2]
```diff
@@ -13,9 +13,10 @@
 
 package tech.pegasys.teku.networking.eth2.peers;
 
+import java.util.Optional;
 import org.apache.tuweni.units.bigints.UInt256;
 import tech.pegasys.teku.networking.p2p.peer.Peer;
 
 public interface DiscoveryNodeIdExtractor {
-  UInt256 calculateDiscoveryNodeId(Peer peer);
+  Optional<UInt256> calculateDiscoveryNodeId(Peer peer);
```
Comment (Nashatyrev) [inline networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/peers/DiscoveryNodeIdExtractor.java:21]: For `Libp2pPeer` the `nodeId` would always exist. We have no other `Peer` implementations aside of libp2p. Do we really need to handle hypothetical case  of absent `NodeId` ?

**Review by Nashatyrev — APPROVED** [review]
LGTM
just a couple of nits for discussion

**Comment by zilm13** [comment]
Decided to not get too far from master and keep it

**Pull request merged**